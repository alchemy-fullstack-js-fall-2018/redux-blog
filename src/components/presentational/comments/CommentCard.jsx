import React from 'react';
import PropTypes from 'prop-types';
import styled from 'styled-components';

const StyledDiv = styled.div`
  margin: 5px;
  border: 1px solid black;
`;

const CommentCard = ({ comment }) => {
  const { id, postId, name, email, body } = comment;

  return (
    <StyledDiv>
      <p>id: {id}</p>
      <p>postId: {postId}</p>
      <p>name: {name}</p>
      <p>email: {email}</p>
      <p>body: {body}</p>
    </StyledDiv>
  );
};

CommentCard.propTypes = {
  comment: PropTypes.object.isRequired
};

export default CommentCard;
