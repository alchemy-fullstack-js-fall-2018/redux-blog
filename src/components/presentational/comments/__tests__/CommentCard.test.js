import React from 'react';
import { shallow } from 'enzyme';
import CommentCard from '../CommentCard.jsx';
import mockComments from '../../../../testing/fixtures/comments.json';

describe('<CommentCard />', () => {
  test('renders', () => {
    const comment = mockComments.filter(comment => comment.postId === 1)[0];
    const wrapper = shallow(
      <CommentCard
        comment={comment}
      />
    );
    expect(wrapper).toMatchSnapshot();
  });
});
