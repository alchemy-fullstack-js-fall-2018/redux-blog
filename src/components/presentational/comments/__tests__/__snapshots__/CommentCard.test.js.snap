// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<CommentCard /> renders 1`] = `
<styled.div>
  <p>
    id: 
    1
  </p>
  <p>
    postId: 
    1
  </p>
  <p>
    name: 
    id labore ex et quam laborum
  </p>
  <p>
    email: 
    Eliseo@gardner.biz
  </p>
  <p>
    body: 
    laudantium enim quasi est quidem magnam voluptate ipsam eos
tempora quo necessitatibus
dolor quam autem quasi
reiciendis et nam sapiente accusantium
  </p>
</styled.div>
`;
